File: sol/1128.cpp
struct Solution {
    int numEquivDominoPairs(vector<vector<int>>& dominoes) {
        int count[100] = {0}, res = 0;
        for (auto &dom : dominoes){
            int a = dom[0], b = dom[1];
            if (a > b){
                swap(a, b);
            }
            res += count[a * 10 + b]++;
        }
        return res;
    }
};

File: sol/1128.py
from collections import defaultdict

class Solution:
    def numEquivDominoPairs(self, dominoes: List[List[int]]) -> int:
        h = defaultdict(int)
        for dom in dominoes:
            a, b = dom
            if a > b:
                a, b = b, a
            h[a * 10 + b] += 1
        return sum(v * (v - 1) // 2 for v in h.values())
